ext/nnstreamer/tensor_filter/tensor_filter_snpe.cc

```diff
@@ -221,17 +227,18 @@ snpe_subplugin::runtimeToString (zdl::DlSystem::Runtime_t runtime)
  * @brief Internal method to get names of output layers from tensors information.
  */
 bool
-snpe_subplugin::set_output_layer_names (const GstTensorsInfo *info)
+snpe_subplugin::set_output_tensor_names (const GstTensorsInfo *info)
 {
+  if (output_tensor_names_list.size () > 0) {
+    output_tensor_names_list = zdl::DlSystem::StringList ();
+  }
   for (unsigned int i = 0; i < info->num_tensors; ++i) {
-    if (info->info[i].name == nullptr) {
+    if (info->info[i].name == nullptr || info->info[i].name[0] == '\0') {
       /* failed */
-      nns_loge ("Given layer name with index %u is invalid.", i);
-      output_layer_names.clear ();
+      nns_loge ("Given output tensor name with index %u is invalid, it is null.", i);
       return false;
     }
-    nns_logd ("Add output layer name of %s", info->info[i].name);
-    output_layer_names.emplace_back (std::string (info->info[i].name));
+    output_tensor_names_list.append (g_strdup (info->info[i].name));
```

Comment:
line 241 and 312
does it need memcpy - strdup()?

Reply:
Thanks! It was memleak. I updated the code